lib/src/pdf.rs

```diff
@@ -416,14 +417,68 @@ impl TransformerTrait for Transformer {
                         )?;
                     }
                 }
+
+                // This currently doesn't support inline, inline support will be added to Paragraph itself.
+                Hyperlink { title, url, alt } => {
+                    let text = title;
+                    let font = pdf.add_builtin_font(BuiltinFont::Courier)?;
+
+                    let font_size = 16_u8; // this is the typographical size,
+                                           // currently it's set to default "16"
+
+                    let font_width = 0.3528 * (font_size as f32) * 0.6;
+
+                    let current_layer = pdf.get_page(*page).get_layer(*layer);
+
+                    *vertical_position += 0.3528 * (font_size as f32);
+
+                    let (x, y) = (
+                        document.left_page_indent + 0.0,
+                        document.page_height - *vertical_position,
+                    );
+
+                    current_layer.use_text(text, font_size as f32, Mm(x), Mm(y), &font);
+
+                    let y = y - 0.3;
+
+                    // Adding the clickable border box around the text.
+                    current_layer.add_link_annotation(LinkAnnotation::new(
+                        printpdf::Rect::new(
+                            Mm(x),
+                            Mm(y),
+                            Mm(x + ((text.len() as f32) * font_width)),
+                            Mm(y + font_width + 0.3),
+                        ),
+                        Some(printpdf::BorderArray::default()),
+                        Some(printpdf::ColorArray::Transparent),
+                        printpdf::Actions::uri(url.clone()),
+                        Some(printpdf::HighlightingMode::Invert),
+                    ));
+
+                    // Insertion of UnderLine (Can be improved)
+                    current_layer.add_link_annotation(LinkAnnotation::new(
+                        printpdf::Rect::new(
+                            Mm(x),
+                            Mm(y),
+                            Mm(x + ((text.len() as f32) * font_width)),
+                            Mm(y),
+                        ),
+                        Some(printpdf::BorderArray::default()),
+                        Some(printpdf::ColorArray::Gray([0.0])),
+                        printpdf::Actions::uri(url.clone()),
+                        Some(printpdf::HighlightingMode::Invert),
+                    ));
+                }
+
                 _ => {}
             }
 
             Ok(())
         }
+
```

Comment:
Look at https://github.com/igumnoff/shiva/issues/4 please

Do you see "2. Units test requered for this new feature"

I do not see any tests about hyperlink feature